elements/pfe-modal/pfe-modal.scss

```diff
@@ -1,154 +1,181 @@
-		padding: pfe-var(container-padding);
-
-		@media screen and (min-width: $pfe-modal--breakpoint--medium) {
-			top: 	pfe-var(container-padding);
-			right: 	pfe-var(container-padding);
-		}
-
-		@media screen and (max-height: $pfe-modal--breakpoint--medium) and (min-width: $pfe-modal--breakpoint--medium) {
-			top: 	calc(#{pfe-var(container-padding)} / 2);
-			right: 	calc(#{pfe-var(container-padding)} / 2);
-		}
-
-		> svg {
-			fill: 	pfe-var(text);
-			height: pfe-var(ui--element--size);
-			width: 	pfe-var(ui--element--size);
-			height: pfe-local(size, $region: close);
-			width: 	pfe-local(size, $region: close);
-		}
-	}
+[hidden] {
+  display: none !important;
 }
 
-#dialog {
-	--pfe-theme--color--text: var(--pfe-broadcasted--text);
-	background-color: var(--pfe-context-background-color);
-	color: var(--pfe-broadcasted--text);
+section {
+  display: flex;
+  position: fixed;
+  height: 100%;
+  width: 100%;
+  top: 0;
+  left: 0;
+  align-items: center;
+  justify-content: center;
+  z-index: var(--pf-c-modal-box--ZIndex,
+    var(--pf-global--ZIndex--xl,
+      500));
+}
+
+#container {
+  position: relative;
+  max-height: inherit;
+}
+
+[part=overlay] {
+  position: fixed;
+  height: 100%;
+  width: 100%;
+  top: 0;
+  left: 0;
+  background-color: var(--pf-c-backdrop--BackgroundColor,
+      var(--pf-global--BackgroundColor--dark-transparent-100,
+        rgba(3, 3, 3, 0.62)));
+}
+
+[part=dialog] {
+  position: relative;
+  margin: 0 auto;
+
+  width: var(--pf-c-modal-box--Width,
+    calc(100% - var(--pf-global--spacer--xl,2rem)));
+
+  max-width: var(--pf-c-modal-box--MaxWidth,
+    calc(100% - var(--pf-global--spacer--xl,2rem)));
+
+  max-height: var(--pf-c-modal-box--MaxHeight,
+    calc(100% - var(--pf-global--spacer--2xl,3rem)));
+
+  box-shadow: var(--pf-c-modal-box--BoxShadow,
+    var(--pf-global--BoxShadow--xl,
+      0 1rem 2rem 0 rgba(3, 3, 3, 0.16),
+      0 0 0.5rem 0 rgba(3, 3, 3, 0.1)));
+
+	background-color: var(--pf-c-modal-box--BackgroundColor,
+    var(--pf-global--BackgroundColor--100,
+      var(--pf-global--BackgroundColor--100, #fff)));
+
+  padding: var(--pf-global--spacer--lg, 1.5rem);
+  margin-inline: var(--pf-global--spacer--md, 1rem);
+}
+
+:host([width]) [part=dialog],
+:host([variant]) [part=dialog] {
+  margin-inline: 0;
+}
+
+:host([width="small"]) [part=dialog],
+:host([variant="small"]) [part=dialog] {
+  --pf-c-modal-box--Width: var(--pf-c-modal-box--m-sm--sm--MaxWidth, 35rem);
+}
+
+:host([width="medium"]) [part=dialog],
+:host([variant="medium"]) [part=dialog] {
+  --pf-c-modal-box--Width: var(--pf-c-modal-box--m-md--Width, 52.5rem);
+}
+
+:host([width="large"]) [part=dialog],
+:host([variant="large"]) [part=dialog] {
+  --pf-c-modal-box--Width: var(--pf-c-modal-box--m-lg--lg--MaxWidth, 70rem);
+}
+
+
+[part=content] {
+  overflow-y: auto;
+  overscroll-behavior: contain;
+
+  max-height: var(--pf-c-modal-box--MaxHeight,
+    calc(100% - var(--pf-global--spacer--2xl,
+      3rem)));
```

Comment:
In testing, it seems the overflow wasn't working correctly.  Here is what I did to fix it.

Add `style="--pf-c-modal-box--MaxWidth: 320px;
    --pf-c-modal-box--Width: 100%;"` to the overflow example demo.  

Then change 100% to 100vh for the max-height calc.

```Suggestion
max-height: var(--pf-c-modal-box--MaxHeight,
    calc(100vh - var(--pf-global--spacer--2xl,
      3rem)));
```